src/NuGet.Services.Metadata.Catalog.Monitoring/Resources/PackageRegistrationMetadataResourceV3.cs

```diff
@@ -24,33 +25,54 @@ public class PackageRegistrationMetadataResourceV3 : IPackageRegistrationMetadat
             _client = client;
         }
 
-        public async Task<PackageRegistrationIndexMetadata> GetIndex(PackageIdentity package, ILogger log, CancellationToken token)
+        public async Task<PackageRegistrationIndexMetadata> GetIndexAsync(PackageIdentity package, ILogger log, CancellationToken token)
         {
-            var feedPackage = await GetPackageFromIndex(package, log, token);
-            return feedPackage != null ? 
-                JsonConvert.DeserializeObject<PackageRegistrationIndexMetadata>(feedPackage.ToString()) : 
-                null;
+            try
+            {
+                var feedPackage = await GetPackageFromIndex(package, log, token);
+                return feedPackage != null ?
+                    JsonConvert.DeserializeObject<PackageRegistrationIndexMetadata>(feedPackage.ToString()) :
+                    null;
+            }
+            catch (Exception e)
+            {
+                throw new ValidationException($"Could not fetch {nameof(PackageRegistrationIndexMetadata)} from V3!", e);
+            }
         }
 
-        public async Task<PackageRegistrationLeafMetadata> GetLeaf(PackageIdentity package, ILogger log, CancellationToken token)
+        public async Task<PackageRegistrationLeafMetadata> GetLeafAsync(PackageIdentity package, ILogger log, CancellationToken token)
         {
-            var feedPackage = await GetPackageFromLeaf(package, log, token);
-            return feedPackage != null ?
-                JsonConvert.DeserializeObject<PackageRegistrationLeafMetadata>(feedPackage.ToString()) :
-                null;
+            try
+            {
+                var feedPackage = await GetPackageFromLeaf(package, log, token);
+                return feedPackage != null ?
+                    JsonConvert.DeserializeObject<PackageRegistrationLeafMetadata>(feedPackage.ToString()) :
+                    null;
+            }
+            catch (Exception e)
+            {
+                throw new ValidationException($"Could not fetch {nameof(PackageRegistrationLeafMetadata)} from V3!", e);
+            }
         }
 
-        private async Task<JObject> GetPackageFromIndex(PackageIdentity package, ILogger log, CancellationToken token)
+        private Task<JObject> GetPackageFromIndex(PackageIdentity package, ILogger log, CancellationToken token)
         {
-            return await _registration.GetPackageMetadata(package, log, token);
+            return _registration.GetPackageMetadata(package, log, token);
```

Comment:
The GetPackageFromIndex and GetPackageFromLeaf are not consistent. One return null on exception cases one bubble up the exception. Is this difference intended?